mypy/newsemanal/semanal_classprop.py

```diff
@@ -95,7 +95,7 @@ def calculate_class_abstract_status(typ: TypeInfo, is_stub_file: bool, errors: E
     if is_stub_file:
         if typ.declared_metaclass and typ.declared_metaclass.type.fullname() == 'abc.ABCMeta':
             return
-        if typ.is_protocol:
+        if typ.is_protocol or typ.typeddict_type:
```

Comment:
I am curious why this is not needed in the old analyzer. Is it an old analyzer bug?

Reply:
I think that the old semantic analyzer just doesn't run this code at all for TypedDicts.

Reply:
I updated the code to be closer to the logic in the old semantic analyzer.